Source/WebCore/css/CSSShapeSegmentValue.cpp

```diff
@@ -199,5 +200,76 @@ String CSSShapeSegmentValue::customCSSText() const
     return builder.toString();
 }
 
-} // namespace WebCore
 
+BasicShapeShape::ShapeSegment CSSShapeSegmentValue::toShapeSegment(const CSSToLengthConversionData& conversionData) const
```

Comment:
Generally, I have been preferring to pass Style::BuilderState& for functions like this, and then using Style::BuilderConverter's helpers where possible. That allows a bit more bottlenecking/shared conversion code (which takes things like zoom into account where sometimes it is forgotten).

Can that be done here?

Reply:
Oh, whoops, I didn't see this was a draft, sorry.

Reply:
I did this, and it caused an SVG zoom bug which I fixed.